### PR TITLE
fix: use the getColumnType and getColumn functions in Graph/view

### PR DESCRIPTION
**Pull request by ChitlangeSahas**
Closes: https://github.com/influxdata/ui/issues/3965
Closes: https://github.com/influxdata/ui/issues/3920

Those functions better deal with the uncertainty of columns not existing, inspired from the Band graph's code. 

Video: 

https://user-images.githubusercontent.com/18511823/155772141-2e2b158b-aa29-495b-b1f0-decc4264e715.mov

### CODE REVIEW

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 56 | :red_circle:
accessibility | 97 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=3967)

File: cypress/e2e/shared/dashboardsIndex.test.ts [hunk 1]
```diff
@@ -627,6 +627,10 @@ describe('Dashboards', () => {
     cy.getByTestID('selector-list p').click()
     cy.getByTestID('time-machine-submit-button').click()
     cy.getByTestID('save-cell--button').click()
+
+    // freeze the clock to a date so we can come back and check screenshots
+    // relative to the same timestamp.
+    cy.clock(newDate)
```
Comment (ChitlangeSahas) [inline cypress/e2e/shared/dashboardsIndex.test.ts:633]: https://przemuh.pl/en/blog/stop-the-time-with-cyclock

Comment (ChitlangeSahas) [reply cypress/e2e/shared/dashboardsIndex.test.ts:633]: The problem was that the test was going back in time to compare the two graphs, but due to render times etc, the past time is not the same as the current. Therefore, the two snapshots won't be the same either.

**Review by hoorayimhelping — APPROVED** [review]
nice usage of `cy.clock`

**Comment by ChitlangeSahas** [comment]
On hold until we know what to do with the snapshot tests.

**Comment by ChitlangeSahas** [comment]
We decided to skip the tests for now until we decide what to do with the Snapshot tests.

**Review by kristinarobinson — APPROVED** [review]

**Pull request merged**